plugins/adr/src/components/AdrReader/decorators.ts

```diff
@@ -37,11 +37,13 @@ export const adrDecoratorFactories = Object.freeze({
   /**
    * Rewrites relative Markdown embeds using absolute URLs.
    */
-  createRewriteRelativeEmbedsDecorator(): AdrContentDecorator {
+  createRewriteRelativeEmbedsDecorator(
+    backendUrl: string,
+  ): AdrContentDecorator {
     return ({ baseUrl, content }) => ({
       content: content.replace(
         /!\[([^\[\]]*)\]\((?!https?:\/\/)(.*?)(\.png|\.jpg|\.jpeg|\.gif|\.webp)(.*)\)/gim,
-        `![$1](${baseUrl}/$2$3$4)`,
+        `![$1](${backendUrl}/image?url=${baseUrl}/$2$3$4)`,
       ),
```

Comment:
Hm I'm not so sure about this - it makes the interface for using this decorator factory a bit confusing for the user. It's a bit unfortunate that the backend url has to be passed here since the user should not have to be concerned about this. It's opaque to the user if they're relying on the default instantiated decorators since this change takes care of calling this decorator factory with the appropriate backend url but someone who is overriding the default and wants to use this will need to emulate this process.

Reply:
Used the  `transformImageUri` from `MarkdownContent` as you suggested instead of modifying this decorator.